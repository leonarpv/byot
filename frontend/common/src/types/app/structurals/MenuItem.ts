import * as React from 'react';

export interface MenuItem<I = string | number> {
  label: string;
  id: I;
  icon?: React.ReactNode;
}

export interface LinkMenuItem<I = string | number> extends MenuItem<I> {
  link: string;
}
