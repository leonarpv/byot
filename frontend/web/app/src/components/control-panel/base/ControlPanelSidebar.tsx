import * as React from 'react';
import {makeStyles, Theme, Box} from '@material-ui/core';
import {ExitToAppRounded} from '@material-ui/icons';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {DrawerMenu} from '../drawer/DrawerMenu';
import {Logo} from '@byot-frontend/web-common/src/components/elementary/logo/Logo';
import {Copyright} from '@byot-frontend/web-common/src/components/content/Copyright';
import {LinkMenuItem} from '@byot-frontend/common/src/types/app/structurals/MenuItem';
import {useTranslation} from 'react-i18next';
import {Router} from '../../../router/Router';

interface Props extends WithStyles<typeof styles> {
  menuItems: LinkMenuItem[];
  selected?: string;
}

const styles = (theme: Theme) => ({
  root: {
    position: 'fixed' as 'fixed',
    left: 0,
    top: 0,
    height: '100%',
    padding: theme.spacing(2),
    display: 'flex' as 'flex',
    flexDirection: 'column' as 'column',
    justifyContent: 'space-between' as 'space-between',
    alignItems: 'center' as 'center',
    width: '100%',
    [theme.breakpoints.between('sm', 'md')]: {
      width: '33%',
    },
    [theme.breakpoints.up('md')]: {
      width: '25%',
    },
  },
});
const useStyles = makeStyles(styles);

export const ControlPanelSidebar: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  const {t} = useTranslation();
  return (
    <div className={styles.root}>
      <Logo height={64} />
      <DrawerMenu menu={props.menuItems} selected={props.selected} />
      <Box width="100%" textAlign="center">
        <DrawerMenu
          disablePadding
          menu={[
            {
              label: t('Log out'),
              link: Router.logout.URI(),
              id: 'logout',
              icon: <ExitToAppRounded />,
            },
          ]}
        />
        <Copyright />
      </Box>
    </div>
  );
};
