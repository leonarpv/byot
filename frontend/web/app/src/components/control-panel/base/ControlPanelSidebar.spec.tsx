import * as React from 'react';
import {render} from '@testing-library/react';
import {ControlPanelSidebar} from './ControlPanelSidebar';
import {LinkMenuItem} from '@byot-frontend/common/src/types/app/structurals/MenuItem';

describe('<ControlPanelSidebar/>', () => {
  it('should render', () => {
    const {container} = render(<ControlPanelSidebar menuItems={[]} />);
    expect(container).toMatchSnapshot();
  });
  it('should render as many items as provided + logout item', () => {
    const menuItems: LinkMenuItem[] = [
      {
        id: 0,
        label: 'Label 0',
        link: '/route-0',
      },
      {
        id: 1,
        label: 'Label 1',
        link: '/route-1',
      },
    ];
    const {container, queryAllByTestId} = render(<ControlPanelSidebar menuItems={menuItems} />);
    expect(queryAllByTestId('control-panel-drawer-menu-item')).toHaveLength(menuItems.length + 1);
    expect(container).toMatchSnapshot();
  });
});
