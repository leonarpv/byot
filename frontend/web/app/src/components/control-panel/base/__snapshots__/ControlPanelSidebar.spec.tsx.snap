// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlPanelSidebar/> should render 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-5 makeStyles-root-2 makeStyles-root-4"
    >
      <img
        class="makeStyles-img-3"
        src="logo.svg"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-7 makeStyles-root-6"
    >
      <ul
        class="MuiList-root"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-8"
    >
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-root-6"
      >
        <ul
          class="MuiList-root"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-10 MuiListItem-gutters MuiListItem-button"
            data-testid="control-panel-drawer-menu-item"
            role="button"
            tabindex="0"
            to="/logout"
          >
            <span
              class="makeStyles-highlight-11 makeStyles-highlight-14"
            />
            <div
              class="MuiListItemIcon-root makeStyles-icon-12 makeStyles-icon-15"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10.79 16.29c.39.39 1.02.39 1.41 0l3.59-3.59c.39-.39.39-1.02 0-1.41L12.2 7.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L12.67 11H4c-.55 0-1 .45-1 1s.45 1 1 1h8.67l-1.88 1.88c-.39.39-.38 1.03 0 1.41zM19 3H5c-1.11 0-2 .9-2 2v3c0 .55.45 1 1 1s1-.45 1-1V6c0-.55.45-1 1-1h12c.55 0 1 .45 1 1v12c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1v3c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-13 makeStyles-textPrimary-16 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Log out
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <p
        class="MuiTypography-root makeStyles-typography-17 MuiTypography-body1"
      >
        All rights reserved to
        <strong>
           byoT
        </strong>
      </p>
    </div>
  </div>
</div>
`;

exports[`<ControlPanelSidebar/> should render as many items as provided + logout item 1`] = `
<div>
  <div
    class="makeStyles-root-18"
  >
    <div
      class="MuiBox-root MuiBox-root-22 makeStyles-root-19 makeStyles-root-21"
    >
      <img
        class="makeStyles-img-20"
        src="logo.svg"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-24 makeStyles-root-23"
    >
      <ul
        class="MuiList-root"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root makeStyles-root-25 MuiListItem-gutters MuiListItem-button"
          data-testid="control-panel-drawer-menu-item"
          role="button"
          tabindex="0"
          to="/route-0"
        >
          <span
            class="makeStyles-highlight-26 makeStyles-highlight-29"
          />
          <div
            class="MuiListItemIcon-root makeStyles-icon-27 makeStyles-icon-30"
          />
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-28 makeStyles-textPrimary-31 MuiTypography-body1 MuiTypography-displayBlock"
            >
              Label 0
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root makeStyles-root-25 MuiListItem-gutters MuiListItem-button"
          data-testid="control-panel-drawer-menu-item"
          role="button"
          tabindex="0"
          to="/route-1"
        >
          <span
            class="makeStyles-highlight-26 makeStyles-highlight-32"
          />
          <div
            class="MuiListItemIcon-root makeStyles-icon-27 makeStyles-icon-33"
          />
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-28 makeStyles-textPrimary-34 MuiTypography-body1 MuiTypography-displayBlock"
            >
              Label 1
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
    <div
      class="MuiBox-root MuiBox-root-35"
    >
      <div
        class="MuiBox-root MuiBox-root-36 makeStyles-root-23"
      >
        <ul
          class="MuiList-root"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-25 MuiListItem-gutters MuiListItem-button"
            data-testid="control-panel-drawer-menu-item"
            role="button"
            tabindex="0"
            to="/logout"
          >
            <span
              class="makeStyles-highlight-26 makeStyles-highlight-37"
            />
            <div
              class="MuiListItemIcon-root makeStyles-icon-27 makeStyles-icon-38"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10.79 16.29c.39.39 1.02.39 1.41 0l3.59-3.59c.39-.39.39-1.02 0-1.41L12.2 7.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L12.67 11H4c-.55 0-1 .45-1 1s.45 1 1 1h8.67l-1.88 1.88c-.39.39-.38 1.03 0 1.41zM19 3H5c-1.11 0-2 .9-2 2v3c0 .55.45 1 1 1s1-.45 1-1V6c0-.55.45-1 1-1h12c.55 0 1 .45 1 1v12c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1v3c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-28 makeStyles-textPrimary-39 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Log out
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <p
        class="MuiTypography-root makeStyles-typography-40 MuiTypography-body1"
      >
        All rights reserved to
        <strong>
           byoT
        </strong>
      </p>
    </div>
  </div>
</div>
`;
